index.bs

```diff
@@ -17,12 +17,20 @@ Include MDN Panels: if possible
 Include Can I Use Panels: yes
 </pre>
 
-Introduction {#intro}
-=====================
+# Introduction # {#introduction}
 
-For now, see the [explainer]([REPOSITORYURL]).
+A document can request the user agent that the default speculative parsing behavior is not used, by using the `Prefer-No-Speculative-Parsing` HTTP header. This header is a HTTP structured field whose value must be Boolean.
 
-See [https://garykac.github.io/procspec/](https://garykac.github.io/procspec/),
-[https://dlaliberte.github.io/bikeshed-intro/index.html](https://dlaliberte.github.io/bikeshed-intro/index.html),
-and [https://speced.github.io/bikeshed/](https://speced.github.io/bikeshed/) to get started on your
-specificaton.
+Per the processing model in the [Speculative HTML parsing](https://html.spec.whatwg.org/multipage/parsing.html#speculative-html-parsing), values that are not the structured header boolean true value (i.e., `?1`) will be ignored.
+
+The consequence of using this header is that the resulting Document's [active speculative HTML parser](https://html.spec.whatwg.org/multipage/parsing.html#active-speculative-html-parser) may be set to `null`. This header indicates only a preference from the Document on how the speculative HTML parser may behave, which the user agent may choose to ignore. In terms of observable effects, this means that any fetches originating from speculative HTML parser may be avoided and any time spent in speculative parsing may be reduced. Resources that were being fetched as part of a speculative fetch, will then be fetched as part of the normal document parsing. Behind the scenes, this preference can allow user agents to skip the time spent in speculative parsing, and deallocate any implementation specific resources corresponding to speculative parsing for additional efficiency.
```

Comment:
This, and the example below, is basically explainer text. We need to have a precise specification instead.

Here is how you would do that:

- Rename this "Intro" section to something like "Modifications to HTML's [Speculative HTML parsing](https://html.spec.whatwg.org/#speculative-html-parsing) section.
- Remove this paragraph and the example below.
- Add a sentence like: `Each {{Document}} has a <dfn for="Document">prefer no speculative HTML parsing</dfn> boolean, initially false.`
- Add a paragraph explaining that you want to modify [start the speculative HTML parser](https://html.spec.whatwg.org/#start-the-speculative-html-parser) by changing step 1 to something like: `Optionally, return. The user agent may especially wish to return if |parser|'s {{Document}}'s [=Document/prefer no speculative HTML parsing=] is true, as that indicates the web developer prefers to opt out of speculative HTML parsing.`
- Add a new section, titled something like "Modifications to `Document` creation".
    - In that section, explain that you want to modify [create and initialize a Document object](https://html.spec.whatwg.org/#initialise-the-document-object), by inserting steps between the current step 9 and 10, which parse the document's header list looking for `Prefer-No-Speculative-Parsing`, and stores the result in a boolean variable `|preferNoSpeculativeHTMLParsing|`. Those steps will look similar to the current spec's step 8.1+8.2.
   - Also in that section, explain that you want to modify the current step 10 to set `|document|'s [=Document/prefer no speculative HTML parsing=]` to `|preferNoSpeculativeHTMLParsing|`.

Remember, the best specifications parallel implementation code. Hopefully you can see how the above is a more detailed example of doing so than the explanatory text you've written here.

Reply:
Thank you! I understood that I had to modify HTML's [Speculative HTML parsing] section, but wasn't clear on how to represent such a patch in a different spec. This makes it super clear. I've revised the PR with the changes add. PTAL!